src/mapgen.cpp

```diff
@@ -361,8 +362,25 @@ void load_mapgen( JsonObject &jo )
         }
     } else if ( jo.has_string( "om_terrain" ) ) {
         load_mapgen_function(jo, jo.get_string( "om_terrain" ), -1);
+    } else if( jo.has_array( "om_terrain_multi" ) ) {
```

Comment:
I suppose you don't need another name. `om_terrain` can be treated either as string or an array of strings (depending on its content). This is also more robust: `om_terrain_multi` won't be silently ignored in favor of `om_terrain`.